File: Algorithm/acmicpc/p1946.py
import sys
input = sys.stdin.readline

for i in range(int(input())):
    N = int(input())
    M = [0 for i in range(N)]
    for i in range(N):
        M[i] = list(map(int,input().split()))
    M.sort()
    
    limit = M[0][1]
    cnt = 1

    for i in range(1,N):
        if limit > M[i][1]:
            cnt += 1
            limit = M[i][1]
    print(cnt)

File: Algorithm/acmicpc/p1946.cpp
#include <iostream>
#include <algorithm>

using namespace std;
int T, N;
struct st {
	int a, b;
}TEMP[100005];
bool cmp(const st &l, const st &r) {
	return l.a < r.a;
}
int main() {
	scanf("%d", &T);
	while (T--) {
		scanf("%d", &N);
		for (int i = 0; i < N; i++) {
			int a, b;
			scanf("%d %d", &a, &b);
			TEMP[i].a = a;
			TEMP[i].b = b;
		}
		sort(TEMP, TEMP + N, cmp);
		int limit = TEMP[0].b;
		int cnt = 1;
		for (int i = 1; i < N; i++) {
			if (limit > TEMP[i].b) {
				cnt++;
				limit = TEMP[i].b;
			}
		}
		printf("%d\n", cnt);
	}
	return 0;
}